src/ibmca-engine-opensslconfig.in

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/perl
+
+#
+# Copyright 2022 International Business Machines Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Generate openssl.cnf from the system config file with added engine section
+# for ibmca engine.
+#
+# USE WITH CARE: No automation can replace human knowledge and understanding
+#                of the desired configuration!
+#
+
+use strict;
+use warnings;
+
+sub generate()
+{
+    my ($osslconfpath);
+    my ($tmp, $ih, $line, $oh, $defaultcnfsect, $indefaultsect, $enginesect);
+
+    $osslconfpath = `openssl version -d` || die "Please install openssl binary";
+    $osslconfpath =~ s/OPENSSLDIR: \"([^\"]*)\"$/$1/ || die "Failed to extract OpenSSL configuration directory";
+    chomp $osslconfpath;
+
+    open($ih, "<", "$osslconfpath/openssl.cnf") or die "Cannot open $osslconfpath/openssl.cnf";
+    open($oh, ">", "openssl.cnf.ibmca") or die "Cannot open openssl.cnf.ibmca";
+
+    $defaultcnfsect = undef;
+    $indefaultsect = 0;
+    $enginesect = undef;
+    while ($line = <$ih>) {
+	if ($line =~ /openssl_conf\s*=\s*(.*)/) {
+	    $defaultcnfsect = $1;
+	    chomp $defaultcnfsect;
+	}
+	if ($indefaultsect) {
+	    if ($line =~ /\[\s*\w+\s*\]/) {
+		if (!$enginesect) {
+		    print $oh "engines = engine_section\n"
```

Comment:
shouldn't you set $enginesect to 'engine_section' here, so that it is set for below code?

Reply:
In this case it should not matter because there was no engine section in the default section.  I keep `$enginesect` undefined and a few lines below I add the engine section.  If `$enginesect` is defined, it should be the name of an existing engine section.  And in that case I do not modify the engine section.  Of course, there can still be problems if the engine section occurs before the default section.  But that is a problem I would like to ignore...

Reply:
ah no, my error, please ignore this comment.